bitrise.yml

```diff
@@ -21,222 +35,146 @@ workflows:
+            npm run build
+  4_A_xcode_build_and_test_Fennec:
+    steps:
+    - git::https://github.com/bitrise-steplib/steps-xcode-build-for-test.git@export-fix:
         inputs:
-        - deploy_path: l10n-screenshots-dd/
-        - is_compress: 'true'
-    - deploy-to-bitrise-io@1.10:
+        - xcodebuild_options: CODE_SIGN_IDENTITY="" CODE_SIGNING_REQUIRED=NO CODE_SIGNING_ALLOWED=NO
+        - scheme: Fennec
+    - xcode-test@2:
         inputs:
-        - deploy_path: l10n-screenshots/en-US/en-US
-        - is_compress: 'true'
-    - cache-push@2.3: {}
-    envs:
-    - opts:
-        is_expand: false
-      BITRISE_SCHEME: L10nSnapshotTest
-    description: >-
-      This Workflow is to run tests (currently SmokeTest) when there is a merge
-      in master
-    meta:
-      bitrise.io:
-        stack: osx-xcode-12.5.x
-        machine_type_id: g2.4core
-  L10nScreenshotsTests:
+        - scheme: Fennec
+        - simulator_device: iPhone 8
+  5_deploy_and_slack:
     steps:
-    - activate-ssh-key@4:
-        run_if: '{{getenv "SSH_RSA_PRIVATE_KEY" | ne ""}}'
-    - git-clone@4: {}
-    - certificate-and-profile-installer@1.10: {}
-    - script@1:
-        title: Pull in L10n
+    - deploy-to-bitrise-io@1.9: {}
+    - cache-push@2.2:
+        is_always_run: true
+    - slack@3.1:
         inputs:
-        - content: |
-            #!/usr/bin/env bash
-            set -e
-            set -x
-            # Import only the shipping locales (from shipping_locales.txt) on Release
-            # builds. Import all locales on Beta and Fennec_Enterprise, except for pull
-            # requests.
-            git clone https://github.com/mozilla-mobile/ios-l10n-scripts.git ||exit 1
-            pip3 install --user virtualenv
-            cd /usr/local/bin
-            ln -s /Users/vagrant/Library/Python/3.9/bin/virtualenv .
-            cd -
-            ./ios-l10n-scripts/import-locales-firefox.sh --release
-    - script@1:
+        - webhook_url: "$WEBHOOK_SLACK_TOKEN"
+  4_B_xcode_build_and_test_Fennec_Enterprise_XCUITests:
+    steps:
+    - git::https://github.com/bitrise-steplib/steps-xcode-build-for-test.git@export-fix:
         inputs:
-        - content: >-
-            #!/usr/bin/env bash
-
-            # fail if any commands fails
-
-            set -e
-
-            # debug log
-
-            set -x
-
-
-            curl --location --retry 5 --output l10n-screenshots-dd.zip
-            "$MOZ_DERIVED_DATA_PATH"
-
-            mkdir l10n-screenshots-dd
-
-            unzip l10n-screenshots-dd.zip -d l10n-screenshots-dd
-
-            rm l10n-screenshots-dd.zip
-        title: Download derived data path
-    - script@1.1:
+        - xcodebuild_options: CODE_SIGN_IDENTITY="" CODE_SIGNING_REQUIRED=NO CODE_SIGNING_ALLOWED=NO
+        - scheme: Fennec_Enterprise_XCUITests
+    - xcode-test@2:
+        inputs:
+        - scheme: Fennec_Enterprise_XCUITests
+        - simulator_device: "$IOS_DEVICE"
+  NewXcodeVersion:
```

Comment:
For this Workflow we can't add the before or after run because running workflows conditionally does not work and would need to execute from the first to the 5th workflow only when the  new xcode version env variable is set to true.

While we found a workaround for that, let's keep the workflow as it is now